synapse/handlers/room.py

```diff
@@ -456,11 +456,19 @@ def _move_aliases_to_new_room(
 
         # check to see if we have a canonical alias.
         canonical_alias = None
+        canonical_alt_aliases = []
         canonical_alias_event_id = old_room_state.get((EventTypes.CanonicalAlias, ""))
         if canonical_alias_event_id:
             canonical_alias_event = yield self.store.get_event(canonical_alias_event_id)
             if canonical_alias_event:
                 canonical_alias = canonical_alias_event.content.get("alias", "")
+                canonical_alt_aliases = canonical_alias_event.content.get(
+                    "alt_aliases", []
+                )
+                # If alt_aliases is not a list, overwrite it as whatever is there
+                # cannot be handled.
+                if not isinstance(canonical_alt_aliases, (list, tuple)):
```

Comment:
Would it make more sense to `unfreeze` this instead of handling it this way?

Reply:
no, that sounds like additional overhead.